# Comparative Genomics
When we se comparative genomics, there can be any kind of coparison between genomes and amulktitude of question we would want answer to. It can also be on multiple taxonomic levels, either species, phyla or even domains of biology could be applied and studied. 

So what do we use comparative genomics for? Phylogeny, so if we want to determine species of a certain organism we would want to observe multiple samples. Molecular evolution is also based on comparative genomics, we often try to identify why some pieces of DNA is either under cselection or different rates of evolution. One could also look at genome dynamics, how the structure of the genome changes over times or how they are different between two groups of organisms. We can also look at conserved elements, which may be regions outside of genes that are under strong selection due to some function. One could also take an epidemiological approachs, to find the origin of an outbreak for instance. 

Comparative genomics is also a powerful tool to connect phenotype to genotype. For instance one could: find genes related to pathegenicity in pathogens, look at species from the same environmental niche and observe convergent or similar solutions for problems in the environment, pathogen diversity in a way that which hosts they may infect, and also observe differences between domesticated vs. wild animals/plants.

To understand relationships in between sequences we need to perform an alignment. But what do we align? Either whole genomes to determine similar parts of the genomes, but if we want to study distant things we need to look at the more conserved functions such as coding regions where we can also choost to either study coding regions in nucleotide or amino acid sequences. 

OrthoMCL is a method to identify clusters of related proteins together, which we then can use to determine which genes are shared between our group of organisms and which are unique. 

Normally things that are shared between the members put into an analyses are called "The Core Genome". This then defines what the group of organisms have in common and what are their function, are also believed to be part of the essential gene pool of the organisms clustered. These Core genes are quite often to be used to create species phylogenies. 

## Phylogenomics
when we use genomes to create phylogeny. We can either use Core genes above to create a phylogenetic tree for every gene and then compare these trees to try and get a representative tree. The more common way is that we make one tree for all of the trees all together, which will investigate each of our genes thus finding wierd genes, where as in the comparison trees we assume that the genes in the core genome are related. This could be true for eukaryotes but in prokaryotes we do find more often HGT. 

## Molecular evolution/selection
We want to determine what is or what is not under positive selection. So genes/proteins that are subject to change. 

## Deciding genome content
Accessory/dispensable genome - genes believed to be essential for different environment either due to local competiation or environemental needs. So this is what is leftover after that the core genome is determined. This part of the genome often contains peticular types of genes, in Eukaryotes and prokaryotes there are mobile elements present, genomic islands and hypothetical proteins(proteins not found before due to high variability and non essential for all lifeforms). 

- Genomic Islands - The reason why this caught interest is that we often look at pathogenicity, and when comparing genomes in bacterium they found particular types of genes and they are thus called genomic islands (GEIs), and in some cases mentioned as Pathogenic Islands(PEIs) due to some genes found in these regions that contribute to teh bacteriums pathogenicity. 

- Pan-genome concept - Important to understand evolutionary history, how often genes are lost and gained in an organism. This can be opened or closed, if opened, then as we sequence more genomes we expand the genome, wheras if closed means that after a certain number of sequences we wont find any more genes. 
